src/java.base/share/lib/security/default.policy

```diff
@@ -103,6 +103,14 @@ grant codeBase "jrt:/jdk.accessibility" {
     permission java.lang.RuntimePermission "accessClassInPackage.sun.awt";
 };
 
+grant codeBase "jrt:/jdk.attach" {
+    permission java.lang.RuntimePermission "accessClassInPackage.com.ibm.tools.attach.target";
+    permission java.lang.RuntimePermission "accessClassInPackage.com.ibm.oti.util";
+    permission java.util.PropertyPermission "com.ibm.tools.attach.*", "read";
+    permission java.io.FilePermission "<<ALL FILES>>", "read,write,delete";
+    permission java.net.SocketPermission "*", "accept,resolve";
```

Comment:
Similar request for this

Reply:
Added comments, and the stacktrace when missing this permission is as following:
```
Exception in thread "main" java.security.AccessControlException: Access denied ("java.net.SocketPermission" "127.0.0.1:45762" "accept,resolve")
	at java.base/java.security.AccessController.throwACE(AccessController.java:176)
	at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:237)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:373)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
	at java.base/java.lang.SecurityManager.checkAccept(SecurityManager.java:940)
	at java.base/java.net.ServerSocket.implAccept(ServerSocket.java:556)
	at java.base/java.net.ServerSocket.accept(ServerSocket.java:519)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.tryAttachTarget(OpenJ9VirtualMachine.java:486)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.attachTargetImpl(OpenJ9VirtualMachine.java:154)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.lambda$attachTarget$1(OpenJ9VirtualMachine.java:129)
	at com.ibm.tools.attach.attacher.OpenJ9VirtualMachine$$Lambda$50.00000000800F0620.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:703)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.attachTarget(OpenJ9VirtualMachine.java:131)
```

Please have another look.

Reply:
I wonder if we really need `"*"`, not sure if `127.0.0.1` is the only address used.

Reply:
This permission is required by `targetServer.accept()` which should only accept localhost connection. `java.net.ServerSocket.accept():si.getInetAddress().getHostAddress()` suggests IP address `127.0.0.1` is sufficient.
Updated the permission accordingly, please have another look.

Reply:
I expect the IPv6 loopback address also needs to be included.
If another machine attempts to connect to the ServerSocket, a security exception will occur from accept(). I think the tryAttachTarget() code could be improved to bind to the loopback address before calling accept(), although perhaps in a separate PR.

Reply:
Changed the address/port to `localhost:0-` as per https://docs.oracle.com/javase/8/docs/api/java/net/SocketPermission.html.
Will look into `ServerSocket.bind()` to loopback address in a separate PR instead.

Please review.

Reply:
As the port is automatically assigned, it should always be in the range `1024-` since the lower port numbers are reserved.

Reply:
Updated the starting port number to `1024`.